### ISSUE TITLE
link-check triggers GoDaddy DoS protection

### ISSUE THREAD

**Issue by dialex** [tcort/markdown-link-check#23]
This [url](http://kaner.com/?p=190) `http://kaner.com/?p=190` is considered as a dead link by your tool, but it's alive and well 🤔

**Comment by dialex** [comment]
Ping. Hello?

**Comment by tcort** [comment]
Your site is returning a `403 Forbidden` error triggered by some DoS protection.

```
{ request:
   { debugId: 1,
     uri: 'http://kaner.com/?p=190',
     method: 'HEAD',
     headers:
      { 'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/47.0.2526.106 Safari/537.36',
        host: 'kaner.com',
        'content-length': 0 } } }
{ response:
   { debugId: 1,
     headers:
      { server: 'nginx',
        date: 'Thu, 01 Feb 2018 14:09:30 GMT',
        'content-type': 'text/html',
        connection: 'close',
        'x-xss-protection': '1; mode=block',
        'x-frame-options': 'SAMEORIGIN',
        'x-content-type-options': 'nosniff',
        'x-sucuri-id': '14008' },
     statusCode: 403,
     body: '' } }
{ request:
   { debugId: 2,
     uri: 'http://kaner.com/?p=190',
     method: 'GET',
     headers:
      { 'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/47.0.2526.106 Safari/537.36',
        host: 'kaner.com' } } }
{ response:
   { debugId: 2,
     headers:
      { server: 'nginx',
        date: 'Thu, 01 Feb 2018 14:09:30 GMT',
        'content-type': 'text/html',
        'transfer-encoding': 'chunked',
        connection: 'close',
        'x-xss-protection': '1; mode=block',
        'x-frame-options': 'SAMEORIGIN',
        'x-content-type-options': 'nosniff',
        'x-sucuri-id': '14008' },
     statusCode: 403,
     body: '<!DOCTYPE html>\n<html lang="en" xmlns="http://www.w3.org/1999/xhtml">\n<head>\n<link rel="stylesheet" href="https://cdn.sucuri.net/css/whitelabel/typography.css?611e1e2" />\n<link rel="stylesheet" href="https://cdn.sucuri.net/css/whitelabel.css?611e1e2" />\n<link rel="stylesheet" href="https://cdn.sucuri.net/css/whitelabel/buttons.css?611e1e2" />\n<link rel="stylesheet" href="https://cdn.sucuri.net/css/whitelabel/footer.css?611e1e2" />\n<link rel="stylesheet" href="https://cdn.sucuri.net/css/whitelabel/header.css?611e1e2" />\n<link rel="stylesheet" href="https://cdn.sucuri.net/css/fonts.css" />\n<meta name="viewport" content="width=device-width, initial-scale=1.0" />\n<title>GoDaddy Security - Access Denied</title>\n</head>\n<body>\n<header class="app-header clearfix">\n<div class="wrap"><a href="https://www.godaddy.com/" class="logo"></a>\n<span class="logo-neartext">Security</span>\n</div>\n</header>\n<section class="center clearfix"></section>\n\n<section class="app-content access-denied clearfix"><div class="box center width-max-940"><h1 class="brand-font font-size-xtra no-margin"><i class="icon-circle-red"></i>Access Denied - GoDaddy Website Firewall</h1>\n<p class="medium-text code-snippet">If you are the site owner (or you manage this site), please whitelist your IP or if you think this block is an error please <a href="https://supportx.sucuri.net/" class="color-green underline">open a support ticket</a> and make sure to include the block details (displayed in the box below), so we can assist you in troubleshooting the issue. </p><h2>Block details:</h1>\n<table class="property-table overflow-break-all line-height-16">\n<tr>\n<td>Your IP:</td>\n<td><span>96.46.34.187</span></td>\n</tr>\n<tr><td>URL:</td>\n<td><span>kaner.com/?p=190</span></td>\n</tr>\n<tr>\n<td>Your Browser: </td>\n<td><span>Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/47.0.2526.106 Safari/537.36</span></td>\n</tr>\n<tr><td>Block ID:</td>\n<td><span>DDOS22</span></td>\n</tr>\n<tr>\n<td>Block reason:</td>\n<td><span>DDOS attempt was blocked.</span></td>\n</tr>\n<tr>\n<td>Time:</td>\n<td><span>Thu Feb  1 09:09:30 2018</span></td>\n</tr>\n<tr>\n<td>Server ID:</td>\n<td><span>14008</span></td></tr>\n</table>\n</div>\n</section>\n\n<footer class="app-footer clearfix"><span>Copyright &copy; 1999 &ndash; 2017 GoDaddy Operating Company, LLC. All rights reserved.</span>\n<span class="padding-left-25"><a href="https://www.godaddy.com/Agreements/Privacy.aspx" class="underline" target="_blank" rel="nofollow noopener">Privacy Policy</a></span>\n</footer>\n</div>\n</body>\n</html>\n' } }
```

**Comment by jmhernandez-circutor** [comment]
I have come across other sites with the same behaviour. Any workaround?

**Comment by NicolasMassart** [comment]
See also #109

**Comment by NicolasMassart** [comment]
closed in favour of #109.
Not that new issue is better, it's just that we only need one and having it on the top of the list. Thanks.
